express/scripts/scripts.js

```diff
@@ -1065,78 +1076,213 @@ export function checkTesting() {
     // reason = 'not prod host';
   }
   if (window.location.hash) {
-    runTest = false;
+    experiment = '';
     // reason = 'suppressed by #';
   }
-  if (window.location.search === '?test') {
-    runTest = true;
-  }
+
   if (navigator.userAgent.match(/bot|crawl|spider/i)) {
-    runTest = false;
+    experiment = '';
     // reason = 'bot detected';
   }
 
-  if (runTest) {
-    let $testTable;
-    document.querySelectorAll('table th').forEach(($th) => {
-      if ($th.textContent.toLowerCase().trim() === 'a/b test') {
-        $testTable = $th.closest('table');
-      }
+  const usp = new URLSearchParams(window.location.search);
+  if (usp.has('experiment')) {
+    [experiment] = usp.get('experiment').split('/');
+  }
+
+  return experiment;
+}
+
+/**
+ * Gets experiment config
+ * @param {string} experiment
+ */
+
+export async function fetchExperimentConfig(experiment) {
+  const path = `/express/experiments/${experiment}/manifest.json`;
+  try {
+    const config = {};
+    const resp = await fetch(path);
+    const json = await resp.json();
+    json.settings.data.forEach((line) => {
+      const key = toCamelCase(line.Name);
+      let value = line.Value;
+      if (key === 'blocks') value = value.split(',').map((e) => e.trim().toLowerCase());
+      config[key] = value;
+    });
+    config.id = experiment;
+
+    const variants = {};
+    let variantNames = Object.keys(json.experiences.data[0]);
+    variantNames.shift();
+    variantNames = variantNames.map((vn) => toCamelCase(vn));
+    variantNames.forEach((variantName) => {
+      variants[variantName] = {};
     });
 
-    const testSetup = [];
-
-    if ($testTable) {
-      $testTable.querySelectorAll('tr').forEach(($row) => {
-        const $name = $row.children[0];
-        const $percentage = $row.children[1];
-        const $a = $name.querySelector('a');
-        if ($a) {
-          const url = new URL($a.href);
-          testSetup.push({
-            url: url.pathname,
-            traffic: parseFloat($percentage.textContent) / 100.0,
-          });
+    let lastKey = 'default';
+
+    json.experiences.data.forEach((line) => {
+      let key = toCamelCase(line.Name);
+      if (!key) key = lastKey;
+      lastKey = key;
+      const vns = Object.keys(line);
+      vns.shift();
+      vns.forEach((vn) => {
+        const camelVN = toCamelCase(vn);
+        if (key === 'pages') {
+          variants[camelVN][key] = variants[camelVN][key] || [];
+          variants[camelVN][key].push(new URL(line[vn]).pathname);
+        } else {
+          variants[camelVN][key] = line[vn];
         }
       });
+    });
+    config.variants = variants;
+    config.variantNames = variantNames;
+    console.log(config);
+    return config;
+  } catch (e) {
+    console.log(`error loading experiment manifest: ${path}`, e);
+  }
+  return null;
+}
+
+/**
+ * replaces main with content from path
+ * @param {string} path
+ */
+
+async function replaceInner(path, main) {
```

Comment:
Nit: I would use `element` instead of `main`. `main` gives the feeling the function is doing something global which we are just talking about one specific element here.